spec/compiler/semantic/automatic_cast.cr

```diff
@@ -468,4 +468,28 @@ describe "Semantic: automatic cast" do
       end
       )) { int64 }
   end
+
+  it "doesn't do multidispatch if an overload matches exactly (#8217)" do
+    assert_type(%(
+      def foo(x : Int64)
+        x
+      end
+
+      def foo(*xs : Int64)
+        xs
+      end
+
+      foo(1)
```

Comment:
how is this checking no multidispatch is done? It comparing positional vs named to me.

Reply:
Because on multidispatch the type would be `Int64 | Tuple(Int64)`.